# GTASSIST
GTASSIST: Objective evaluation measures of genetic marker selection in large-scale SNP genotyping
Kaminuma E, Masuya H, Miura I, Motegi H, Takahasi KR, Nakazawa M, Matsui M, Gondo Y, Noda T, Shiroishi T, Wakana S, Toyoda T.
J Bioinform Comput Biol. 2008 6:905-17.
http://www.ncbi.nlm.nih.gov/pubmed/18942158

High-throughput single nucleotide polymorphism (SNP) genotyping systems provide two kinds of fluorescent 
signals detected from different alleles. In current technologies, the process of genotype discrimination 
requires subjective judgments by expert operators, even when using clustering algorithms. We proposed 
two evaluation measures to manage fluorescent scatter data with nonclear plot aggregation. The first is 
the marker ranking measure, which provides a ranking system for the SNP markers based on the distance 
between the scatter plot distribution and a user-defined ideal distribution. The second measure, called 
individual genotype membership, uses the membership probability of each genotype related to an individual 
plot in the scatter data. In verification experiments, the marker ranking measure determined the ranking 
of SNP markers correlated with the subjective order of SNP markers judged by an expert operator. The 
experiment using the individual genotype membership measure clarified that the total number of unclassified 
individuals was remarkably reduced compared to that of manually unclassified ones. These two evaluation 
measures were implemented as the GTAssist software. GTAssist provides objective standards and avoids 
subjective biases in SNP genotyping workflows.
